src/Compatibility/Microsoft.DotNet.ApiCompatibility/DiagnosticBag.cs

```diff
@@ -28,15 +28,18 @@ public DiagnosticBag(string noWarn, (string diagnosticId, string referenceId)[]
             _noWarn = new HashSet<string>(noWarn?.Split(';'));
             _ignore = new Dictionary<string, HashSet<string>>();
 
-            foreach ((string diagnosticId, string referenceId) in ignoredDifferences)
+            if (ignoredDifferences != null)
             {
-                if (!_ignore.TryGetValue(diagnosticId, out HashSet<string> members))
+                foreach ((string diagnosticId, string referenceId) in ignoredDifferences)
                 {
-                    members = new HashSet<string>();
-                    _ignore.Add(diagnosticId, members);
-                }
+                    if (!_ignore.TryGetValue(diagnosticId, out HashSet<string> members))
+                    {
+                        members = new HashSet<string>();
+                        _ignore.Add(diagnosticId, members);
+                    }
 
-                members.Add(referenceId);
+                    members.Add(referenceId);
+                }
```

Comment:
Are these ApiCompatibility changes expected to be in this PR?

Reply:
yes we were missing a null check.